src/zc/buildout/easy_install.py

```diff
@@ -1564,3 +1540,72 @@ class IncompatibleConstraintError(zc.buildout.UserError):
     """
 
 IncompatibleVersionError = IncompatibleConstraintError # Backward compatibility
+
+
+def _move_to_eggs_dir_and_compile(dist, dest):
+    """Move distribution to the eggs destination directory.
+
+    And compile the py files, if we have actually moved the dist.
+
+    Its new location is expected not to exist there yet, otherwise we
+    would not be calling this function: the egg is already there.  But
+    the new location might exist at this point if another buildout is
+    running in parallel.  So we copy to a temporary directory first.
+    See discussion at https://github.com/buildout/buildout/issues/307
+
+    We return the new location.
+    """
+    # First make sure the destination directory exists.  This could suffer from
+    # the same kind of race condition as the rest: if we check that it does not
+    # exist, and we then create it, it will fail when a second buildout is
+    # doing the same thing.
+    try:
+        os.makedirs(dest)
```

Comment:
I added this `makedirs` line to avoid a failure in a test that calls easy_install with a destination directory that does not exist: https://github.com/buildout/buildout/blob/2.5.2/src/zc/buildout/easy_install.txt#L209-L223
Seems a harmless side effect of this pull request.